code/game/machinery/_machines_base/stock_parts/access_lock.dm

```diff
@@ -46,35 +57,101 @@
 /obj/item/stock_parts/access_lock/attackby(obj/item/W, mob/user)
 	var/obj/machinery/machine = loc
 	if(!emagged && istype(machine))
-		if(check_access(W))
+		var/obj/item/card/id/I = W.GetIdCard()
+		if(I && check_access(I))
 			locked = !locked
 			visible_message(SPAN_NOTICE("\The [src] beeps and flashes green twice: it is now [locked ? "" : "un"]locked."))
 			return TRUE
 		return
+	return ..()
 
-	if(isMultitool(W)) // This is a non-machine interaction only; doing this within a machine will likely lead to interaction clashes.
-		var/list/accesses = user.GetAccess()
-		var/list/choices = list()
-		for(var/key in accesses)
-			choices[get_access_desc(key)] = key
-		choices["No Access"] = "No Access"
-		var/default = length(req_access) ? req_access[1] : null
-		var/selected_desc = input(user, "Select the access you would like \the [src] to use:", "Access Modification", default) as null|anything in choices
-		. = TRUE
-		if(!CanPhysicallyInteract(user))
-			return
-		if(!selected_desc)
-			return
-		if(selected_desc == "No Access")
-			req_access.Cut()
-			return
-		if(!(choices[selected_desc] in user.GetAccess()))
-			return
-		req_access = list(choices[selected_desc])
-		return
+/obj/item/stock_parts/access_lock/attack_self(mob/user)
+	if (!ishuman(user) && !istype(user,/mob/living/silicon/robot))
+		return ..(user)
```

Comment:
Please no. Check fine tool user or whatever if you really think it's needed.

Reply:
Axed